codec-http/src/main/java/io/netty/handler/codec/http/multipart/DiskFileUpload.java

```diff
@@ -157,8 +160,7 @@ protected String getBaseDirectory() {
 
     @Override
     protected String getDiskFilename() {
-        File file = new File(filename);
-        return file.getName();
+        return String.valueOf(fileNameCounter.incrementAndGet());
```

Comment:
The only usage of getDiskFileName is in AbstracktDiskHttpData.java:

    /**
     * @return a new Temp File from getDiskFilename(), default prefix, postfix and baseDirectory
     */
    private File tempFile() throws IOException {
        String newpostfix;
        String diskFilename = getDiskFilename();
        if (diskFilename != null) {
            newpostfix = '_' + diskFilename;
        } else {
            newpostfix = getPostfix();
        }
        File tmpFile;
        if (getBaseDirectory() == null) {
            // create a temporary file
            tmpFile = File.createTempFile(getPrefix(), newpostfix);
        } else {
            tmpFile = File.createTempFile(getPrefix(), newpostfix, new File(
                    getBaseDirectory()));
        }
        if (deleteOnExit()) {
            tmpFile.deleteOnExit();
        }
        return tmpFile;
    }

From what I can see that ensures uniqueness and marks the file for deletion when the JVM exists.